### DISCUSSION TITLE
MATLAB compatibility?

### DISCUSSION THREAD

**Discussion by maciejka3000** [vosen/ZLUDA#134] [Q&A]
Hey, does anyone had success with launching ZLUDA with MATLAB on Windows? After executing command, it immediately halts.

**Comment by maciejka3000** [comment]
Okay, i managed to launch it with `-wait` parameter. GPU is seen properly after using `gpuDevice` command. However, deep learning in MATLAB is not working. I've got this error message:

`Call to cublasDgemm in CUBLAS failed with error status: CUBLAS_STATUS_NOT_SUPPORTED.`

**Reply by pegeac** [reply]
Hi, I'm sorry, I'm not good with this stuff. How did you manage to make it work?
It was windows/Linux? 
Can you write the modified lines you used to make it work?
Moreover, did you use the gpuDevice on MATLAB?

**Reply by maciejka3000** [reply]
Hi, sorry for late reply.
I've tried using ZLUDA on Windows by launching it using following command:
`.\zluda.exe -- matlab -wait`

I had used gpuDevice on matlab r2023a and it showed proper information about GPU. However after updating to r2024a matlab immediately crashes.

**Comment by lshqqytiger** [comment]
Could you try [my fork](https://github.com/lshqqytiger/ZLUDA/releases/tag/v3.5-win)?

**Reply by maciejka3000** [reply]
Hey, thanks for answer. Unfortunately, your solution doesn't work. I've tested your fork with those commands:

- [`trainNetwork`](https://uk.mathworks.com/help/deeplearning/ref/trainnetwork.html)  gives error `An unexpected error occurred trying to launch a kernel. The CUDA error was: named symbol not found`
- [`classify`](https://uk.mathworks.com/help/deeplearning/ref/seriesnetwork.classify.html)  gives error `Unexpected error calling cuDNN: CUDNN_STATUS_EXECUTION_FAILED.`
- [`trainAutoencoder`](https://uk.mathworks.com/help/deeplearning/ref/trainautoencoder.html?s_tid=doc_ta) gives error `Call to cublasDgemm in CUBLAS failed with error status: CUBLAS_STATUS_NOT_SUPPORTED.`

I've also tried multiplying 2 gpu arrays together. Creating gpu variables went fine, but multiplying them gave an error associated with CUBLAS.
